screens/ProductionScreens/BudgetAddExpense.jsx

```diff
@@ -100,13 +111,80 @@ export default function BudgetAddExpense({ navigation, route }) {
+        try {
+            const prodSnapshot = await get(dbRef(db, `productions/${productionCode}/budgets`));
+            if (!prodSnapshot.exists()) {
+                return {};
+            }
+            let newBudgets = {};
+            const budgetUUIDs = Object.keys(prodSnapshot.val());
+            await Promise.all(
+                budgetUUIDs.map(async (budgetUUID) => {
+                    const budgetInfo = await get(dbRef(db, `budgets/${budgetUUID}`));
+                    if (!budgetInfo.exists()) {
+                        return;
+                    }
+                    if (
+                        budgetInfo.val().participants &&
+                        Object.keys(budgetInfo.val().participants).includes(auth.currentUser.uid)
+                    ) {
+                        newBudgets[budgetUUID] = budgetInfo.val();
+                    }
+                })
+            );
+            return newBudgets;
+        } catch (error) {
+            console.log(error.message);
+            setAlert("Could not find all the budgets", "bg-red-400");
+            return {};
+        }
+    };
+
+    const handleChooseBudgetPress = async () => {
+        setModal(
+            <ViewBudgetModal
+                budgets={{}}
+                loading={true}
+                onPress={(budget) => {
+                    setBudget(budget);
+                    setModal(null);
+                }}
+            />
+        );
+        const newBudgets = await fetchBudgets();
+        setPlayBudgets(newBudgets);
+        setModal(
+            <ViewBudgetModal
+                budgets={newBudgets}
+                loading={false}
+                onPress={(budget) => {
+                    setBudget(budget);
+                    setModal(null);
+                }}
+            />
+        );
+    };
+
     return (
         <View className="py-2 flex justify-center items-center">
             <Title extraClassName="mb-2">Add Expense</Title>
-            <ScrollView className="h-5/6 w-full px-8 ">
-                <View className="h-40">
-                    {/* TODO: Select Budget */}
+            <ScrollView className="h-5/6 w-full px-8">
+                <View className="h-40 items-center justify-center">
                     <Text className="text-lg font-semibold text-center">Select Budget</Text>
+                    {budget ? (
+                        <BudgetInfo
+                            budget={playBudgets[budget]}
+                            onClick={handleChooseBudgetPress}
+                        />
+                    ) : (
+                        <TouchableOpacity
+                            onPress={handleChooseBudgetPress}
+                            className="flex bg-neutral-100 p-3 border-2 rounded-lg my-2 w-full h-28 items-center justify-center"
+                        >
+                            <Text className="text-lg">Choose a Budget</Text>
+                        </TouchableOpacity>
```

Comment:
very minor thing but I wish this modal was a few mm wider just because you can slightly see the borders of the form fields behind it - think it might be fixed by the modal itself tho